airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockDestinationDirectLoader.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.mock_integration_test
+
+import io.airbyte.cdk.load.command.DestinationCatalog
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.data.ObjectValue
+import io.airbyte.cdk.load.message.DestinationRecordRaw
+import io.airbyte.cdk.load.mock_integration_test.MockStreamLoader.Companion.getFilename
+import io.airbyte.cdk.load.test.util.OutputRecord
+import io.airbyte.cdk.load.write.DirectLoader
+import io.airbyte.cdk.load.write.DirectLoaderFactory
+import jakarta.inject.Singleton
+import java.time.Instant
+import java.util.UUID
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+
+@Singleton
+class MockDestinationDirectLoaderFactory(
+    private val catalog: DestinationCatalog,
+) : DirectLoaderFactory<MockDestinationDirectLoader> {
+    override fun create(streamDescriptor: DestinationStream.Descriptor, part: Int) =
+        MockDestinationDirectLoader(catalog.getStream(streamDescriptor))
+}
+
+class MockDestinationDirectLoader(
+    private val stream: DestinationStream,
+) : DirectLoader {
+    override fun accept(record: DestinationRecordRaw): DirectLoader.DirectLoadResult {
+        val recordAirbyteValue = record.asDestinationRecordAirbyteValue()
+        val filename = getFilename(record.stream.descriptor, staging = true)
+        val outputRecord =
+            OutputRecord(
+                UUID.randomUUID(),
+                Instant.ofEpochMilli(recordAirbyteValue.emittedAtMs),
+                Instant.ofEpochMilli(System.currentTimeMillis()),
+                stream.generationId,
+                recordAirbyteValue.data as ObjectValue,
+                OutputRecord.Meta(
+                    changes = recordAirbyteValue.meta?.changes ?: listOf(),
+                    syncId = stream.syncId
+                ),
+            )
+        // blind insert into the staging area. We'll dedupe on commit.
+        MockDestinationBackend.insert(filename, outputRecord)
+
+        // HACK: This destination is too fast and causes a race
+        // condition between consuming and flushing state messages
+        // that causes the test to fail. This would not be an issue
+        // in a real sync, because we would always either get more
+        // data or an end-of-stream that would force a final flush.
+        runBlocking { delay(100L) }
```

Comment:
I copied this from the old impl - it's still true; removing the delay causes some tests to hang